pkg/artifactory/resource_artifactory_ldap_setting.go

```diff
@@ -0,0 +1,286 @@
+			"email_attribute": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateIsEmail,
+				Description:  "An attribute that can be used to map a user's email address to a user created automatically in Artifactory.",
+			},
+			"ldap_poisoning_protection": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Protects against LDAP poisoning by filtering out users exposed to vulnerabilities.",
+			},
+			"allow_user_to_access_profile": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Auto created users will have access to their profile page and will be able to perform actions such as generating an API key.",
+			},
+			"paging_support_enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "When set, supports paging results for the LDAP server. This feature requires that the LDAP server supports a PagedResultsControl configuration.",
+			},
+			"search_filter": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: "A filter expression used to search for the user DN used in LDAP authentication. This is an LDAP search filter (as defined in 'RFC 2254') with optional arguments. In this case, the username is the only argument, and is denoted by '{0}'. Possible examples are: (uid={0}) - This searches for a username match on the attribute. Authentication to LDAP is performed from the DN found if successful.",
+			},
+			"search_base": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: "(Optional) A context name to search in relative to the base DN of the LDAP URL. For example, 'ou=users' With the LDAP Group Add-on enabled, it is possible to enter multiple search base entries separated by a pipe ('|') character.",
+			},
+			"search_sub_tree": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "When set, enables deep search through the sub tree of the LDAP URL + search base.",
+			},
+			"manager_dn": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: "The full DN of the user that binds to the LDAP server to perform user searches. Only used with \"search\" authentication.",
+			},
+			"manager_password": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The password of the user that binds to the LDAP server to perform the search. Only used with \"search\" authentication.",
+				Sensitive:   true,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceLdapSettingsRead(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	rClient := m.(*resty.Client)
+	ldapConfigs := &xmlLdapConfig{}
+	rsrcLdapsetting := unpackLdapSetting(d)
+
+	resp, err := rClient.R().Get("artifactory/api/system/configuration")
```

Comment:
I'm not super into prefixing variable name with characters that doesn't provide much value, e.g. `rClient` doesn't do much here. `client` would suffice. Same with `rsrcLdapsetting`. `ldapSetting` works as well and less confusing.

Also, please camelCase `rsrcLdapsetting` correctly.

Reply:
oh, yeah... - this isn't c++ with `sPtr_myvar`